src/nifgen/examples/nifgen_script.py

```diff
@@ -0,0 +1,153 @@
+        session.script_triggers[0].digital_edge_script_trigger_edge = nifgen.ScriptTriggerDigitalEdgeEdge.RISING  # RISING / FAILING
+        # PFI TRIGGER: not used right now
+        # session.script_triggers[1].script_trigger_type = nifgen.script_triggers[0].ScriptTriggerType.DIGITAL_EDGE
+        # session.script_triggers[1].digital_edge_script_trigger_source = 'PFI0'
+        # session.script_triggers[1].digital_edge_script_trigger_edge = session.script_triggers[0].ScriptTriggerDigitalEdgeEdge.RISING
+
+        # 3-Generate Signal
+        sine = generate_sinewave()
+        ramp_up = generate_rampup()
+        ramp_down = generate_ramdown()
+        square = generate_square()
+        triangle = generate_triangle()
+        noise = generate_gaussian_noise()
+
+        # 4-Writes data to the waveform in onboard memory.
+        session.channels[channel].write_waveform('sine', sine)        # (waveform_name, data)
+        session.channels[channel].write_waveform('rampup', ramp_up)
+        session.channels[channel].write_waveform('rampdown', ramp_down)
+        session.channels[channel].write_waveform('square', square)
+        session.channels[channel].write_waveform('triangle', triangle)
+        session.channels[channel].write_waveform('noise', noise)
+
+        # 5-Write Script
+        script_sine = 'script Script\nrepeat until scriptTrigger0\nGenerate sine \nend repeat\nend script'
+        script_ramp_up = 'script Script\nrepeat until scriptTrigger0\nGenerate rampup\nend repeat\nend script'
+        script_ramp_down = 'script Script\nrepeat until scriptTrigger0\nGenerate rampdown\nend repeat\nend script'
+        script_square = 'script Script\nrepeat until scriptTrigger0\nGenerate square\nend repeat\nend script'
+        script_triangle = 'script Script\nrepeat until scriptTrigger0\nGenerate triangle\nend repeat\nend script'
+        script_noise = 'script Script\nrepeat until scriptTrigger0\nGenerate noise \nend repeat\nend script'
+        script_multi = 'script Script\nrepeat until scriptTrigger0\nGenerate triangle\nend repeat\nrepeat until scriptTrigger0\nGenerate sine\nend repeat\nend script'
+
+        # 6-Script to generate
+        ''' SINE / SQUARE / TRIANGLE / RAMP_UP / RAMP_DOWN / NOISE / MULTI '''
+        if shape == 'SINE':
+            script = script_sine
+        elif shape == 'RAMP_UP':
+            script = script_ramp_up
+        elif shape == 'RAMP_DOWN':
+            script = script_ramp_down
+        elif shape == 'SQUARE':
+            script = script_square
+        elif shape == 'TRIANGLE':
+            script = script_triangle
+        elif shape == 'NOISE':
+            script = script_noise
+        elif shape == 'MULTI':
+            script = script_multi
+        else:
+            print('Unknown shape: {}'.format(shape))
+            sys.exit(1)
+
+        session.channels[channel].write_script(script)
```

Comment:
FYI: In NI-FGEN, if you assign different names to the scripts, you can just write them all.